### DISCUSSION TITLE
WMC9999 "Cannot find type System.Security.AccessControl.AccessRule in module mscorlib.dll" when using Microsoft.PowerShell.5.ReferenceAssemblies with Windows App SDK 0.8.3

### DISCUSSION THREAD

**Discussion by adrianghc** [microsoft/WindowsAppSDK#1422] [Q&A] [answered]
I'm having my first go at a small project using the Windows App SDK. I'm following [these instructions](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=desktop-csharp) for creating a WinUI 3 app with Desktop C#. My project needs to execute some PowerShell cmdlets, but I don't want the user to need to install the latest version of PowerShell for that, so I'm just going to make use of Windows PowerShell 5.1 in my project. To that end, I've installed the NuGet package [`Microsoft.PowerShell.5.ReferenceAssemblies`](https://www.nuget.org/packages/Microsoft.PowerShell.5.ReferenceAssemblies/1.1.0). However, when building my project I get this error:

| Code | Description | File | Line |
|---|---|---|---|
| WMC999 | Cannot find type System.Security.AccessControl.AccessRule in module mscorlib.dll | Microsoft.UI.Xaml.Markup.Compiler.interop.targets | 618 |

My code is extremely minimal so far, just a few lines in `App.xaml.cs`, including these `using` statements:

```C#
using System.Collections.ObjectModel;
using System.Management.Automation;
```

I'm not going to assume that this is a bug, but that I'm doing something wrong. However, looking up this error online hasn't really yielded anything helpful, and I'm at a loss as to what I could try next. I'd be very glad about ideas! :)

**Comment by BenJKuhn** [comment, marked as answer]
.NET folks tell me you likely just need to add a reference to this package (v5.0):
https://www.nuget.org/packages/System.Security.AccessControl/

Thanks,

Ben

**Reply by adrianghc** [reply]
This seems to indeed have done the trick! It can be so simple sometimes. :) Thanks a lot! Of course now I get another problem, which is that when calling `PowerShell.Create()`, I get a `Common Language Runtime detected an invalid program.` exception at runtime. But I guess that's a story/issue for another time. 😅
